lib/gui/app.js

```diff
@@ -94,7 +94,8 @@ app.run((AnalyticsService, ErrorService, UpdateNotifierService, SelectionStateMo
   const currentVersion = packageJSON.version;
   const currentReleaseType = release.getReleaseType(currentVersion);
   const shouldCheckForUpdates = UpdateNotifierService.shouldCheckForUpdates({
-    ignoreSleepUpdateCheck: currentReleaseType !== release.RELEASE_TYPE.PRODUCTION
+    ignoreSleepUpdateCheck: currentReleaseType !== release.RELEASE_TYPE.PRODUCTION,
+    lastSleptUpdateNotify: settings.get('lastSleptUpdateNotify')
```

Comment:
Sounds good. Regarding the second suggestion, I propose going the opposite direction: move the actual logic inside the `shouldCheckForUpdates` function, so we can unit test it there (and remove the meaningless `ignoreSleepUpdateChec` option)